### PR TITLE
Check the return value of asprintf

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: src/noit_check_lmdb.c [hunk 1]
```diff
@@ -1164,15 +1166,17 @@ noit_check_lmdb_convert_one_xml_check_to_lmdb(mtev_conf_section_t section, char
...
       return -1;
     }
   }
 
   if(!MYATTR(stringbuf, module, module, sizeof(module))) {
     if(!INHERIT(stringbuf, module, module, sizeof(module))) {
       if(!deleted) {
-        asprintf(error, "check uuid '%s' has no module", uuid_str);
+        if(asprintf(error, "check uuid '%s' has no module", uuid_str) < 0)
+          *error = strdup("check as no module");
```
Comment (pamaddox) [inline src/noit_check_lmdb.c:1179]: Typo - "check has no module"

File: src/noit_check_lmdb.c [hunk 2]
```diff
@@ -1254,16 +1259,19 @@ noit_check_lmdb_convert_one_xml_check_to_lmdb(mtev_conf_section_t section, char
...
-      asprintf(error, "check uuid '%s' has oncheck and period", uuid_str);
+      if(asprintf(error, "check uuid '%s' has oncheck and period", uuid_str) < 0)
+        *error = strdup("check has both oncheck and period");
       return -1;
     }
     if(!MYATTR(int32, timeout, &timeout)) {
       if(!INHERIT(int32, timeout, &timeout)) {
-        asprintf(error, "check uuid '%s' has no timeout", uuid_str);
+        if(asprintf(error, "check uuid '%s' has no timeout", uuid_str) < 0)
+          *error = strdup("check as no timeout");
```
Comment (pamaddox) [inline src/noit_check_lmdb.c:1274]: Typo - "check has no timeout"

Comment (postwait) [inline src/noit_check_lmdb.c:1274]:
```suggestion
          *error = strdup("check has no timeout");
```

File: src/noit_check_lmdb.c (hunk 1 above)
Comment (postwait) [inline src/noit_check_lmdb.c:1179]:
```suggestion
          *error = strdup("check has no module");
```

**Pull request merged**